tests/ovn.at

```diff
@@ -9712,6 +9739,60 @@ reset_pcap_file hv1-vif2 hv1/vif2
 rm -f 1.expected
 rm -f 2.expected
 
+
+# Add one more DNS record to the ls1.
+ovn-nbctl --wait=hv set Logical_switch ls1 dns_records="$DNS1 $DNS2 $DNS3"
+echo "*************************"
+ovn-sbctl list DNS
+echo "*************************"
+ovn-nbctl list DNS
+echo "*************************"
+
+# Test PTR record for IPv4 address using IPv4 packet.
+set_dns_params vm1_ipv4_ptr
+src_ip=`ip_to_hex 10 0 0 4`
+dst_ip=`ip_to_hex 10 0 0 1`
+dns_reply=1
+test_dns 1 f00000000001 f000000000f0 $src_ip $dst_ip $dns_reply $dns_req_data $dns_resp_data
+
+# NXT_RESUMEs should be 11.
+OVS_WAIT_UNTIL([test 11 = `cat ofctl_monitor*.log | grep -c NXT_RESUME`])
+
+$PYTHON "$ovs_srcdir/utilities/ovs-pcap.in" hv1/vif1-tx.pcap > 1.packets
+cat 1.expected | cut -c -48 > expout
+AT_CHECK([cat 1.packets | cut -c -48], [0], [expout])
```

Comment:
Hi @numansiddique, I've added tests, but in spite of DNS feature works well, tests don't pass 👎  (fail on this line).
Could you please take a look to help me understand what am I doing wrong? Thanks in advance.

Tests fail like this:
```
*************************
_uuid               : c338ab88-07f8-4a11-a249-df73240b414e
datapaths           : [35aa2069-f329-43fb-adad-a119711c3197]
external_ids        : {dns_id="93bc8e4c-af7a-4310-b8c2-a917e05aa784"}
records             : {vm3.ovn.org="40.0.0.4"}
_uuid               : fb04bb2d-4d40-4686-88de-336aec79efcc
datapaths           : [35aa2069-f329-43fb-adad-a119711c3197]
external_ids        : {dns_id="3bae3357-b6b4-45f0-9cca-6df67e483b99"}
records             : {vm1.ovn.org="10.0.0.4 aef0::4"}
_uuid               : 6eb07752-2a14-46e3-8620-ac945c11a64f
datapaths           : [35aa2069-f329-43fb-adad-a119711c3197]
external_ids        : {dns_id="71342f26-dea8-4275-bda7-9779caee3142"}
records             : {"4.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.f.e.a.ip6.arpa"=vm1.ovn.org, "4.0.0.10.in-addr.arpa"=vm1.ovn.org}
*************************
_uuid               : 93bc8e4c-af7a-4310-b8c2-a917e05aa784
external_ids        : {}
records             : {vm3.ovn.org="40.0.0.4"}
_uuid               : 71342f26-dea8-4275-bda7-9779caee3142
external_ids        : {}
records             : {"4.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.f.e.a.ip6.arpa"=vm1.ovn.org, "4.0.0.10.in-addr.arpa"=vm1.ovn.org}
_uuid               : 3bae3357-b6b4-45f0-9cca-6df67e483b99
external_ids        : {}
records             : {vm1.ovn.org="10.0.0.4 aef0::4"}
*************************
ovn.at:9477: waiting until test 11 = `cat ofctl_monitor*.log | grep -c NXT_RESUME`...
ovn.at:9477: wait succeeded immediately
./ovn.at:9481: cat 1.packets | cut -c -48
--- expout	2021-02-12 14:50:06.572117297 +0300
+++ /builddir/build/BUILD/ovn-20.06.2/tests/testsuite.dir/at-groups/79/stdout	2021-02-12 14:50:06.572117297 +0300
@@ -1 +0,0 @@
-f00000000001f000000000f00800450000d7000000008011
ofctl_monitor0.log:
> NXT_PACKET_IN2 (xid=0x0): cookie=0x6190c6e3 total_len=71 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x6190c6e3 total_len=71 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x6190c6e3 total_len=71 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:02,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x6190c6e3 total_len=71 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:02,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x2c0db78d total_len=71 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x2c0db78d total_len=69 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=69 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x2c0db78d total_len=69 reg11=0x2,reg12=0x3,reg13=0x4,reg14=0x2,metadata=0x1,in_port=2 (via action) data_len=69 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=4
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:02,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.6,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x6190c6e3 total_len=71 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=71 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x6190c6e3 total_len=71 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=71 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x6190c6e3 total_len=91 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=91 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp6,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,ipv6_src=aef0::4,ipv6_dst=aef0::1,ipv6_label=0x00000,nw_tos=0,nw_ecn=0,nw_ttl=255,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x6190c6e3 total_len=91 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=91 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp6,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,ipv6_src=aef0::4,ipv6_dst=aef0::1,ipv6_label=0x00000,nw_tos=0,nw_ecn=0,nw_ttl=255,tp_src=37428,tp_dst=53 udp_csum:0
> NXT_PACKET_IN2 (xid=0x0): cookie=0x6190c6e3 total_len=132 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=132 (unbuffered)
>  userdata=00.00.00.06.00.00.00.00.00.01.de.10.00.00.00.64
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
> send: NXT_RESUME (xid=0x0): cookie=0x6190c6e3 total_len=132 reg11=0x2,reg12=0x3,reg13=0x1,reg14=0x1,metadata=0x1,in_port=1 (via action) data_len=132 (unbuffered)
>  continuation.bridge=49b36ff1-0f79-48ba-8f46-4f831bce5e99
>  continuation.actions=unroll_xlate(table=0, cookie=0),resubmit(,25)
>  continuation.odp_port=3
> udp,vlan_tci=0x0000,dl_src=f0:00:00:00:00:01,dl_dst=f0:00:00:00:00:f0,nw_src=10.0.0.4,nw_dst=10.0.0.1,nw_tos=0,nw_ecn=0,nw_ttl=128,tp_src=37428,tp_dst=53 udp_csum:0
/builddir/build/BUILD/ovn-20.06.2/tests/testsuite.dir/079/cleanup: line 1: kill: (20020) - No such process
79. ovn.at:8968: 79. ovn -- dns lookup : 1 HV, 2 LS, 2 LSPs/LS (ovn.at:8968): FAILED (ovn.at:9481)
```